### ISSUE TITLE
Can the dbvm_cloak_activate function specify the address length?

### ISSUE THREAD

**Issue by DarkAsm** [cheat-engine/cheat-engine#1981]
@cheat-engine

**Comment by DarkAsm** [comment]
Is dbvm_cloak_activate and shadow walker the same technology? Is it possible to specify the address length? 4096 bytes will take up a lot of cpu, I want to use it as a traceless hook, but I don’t know how to specify the address length.

**Comment by cheat-engine** [comment]
On intel it might be possible to implement, as it has sectioned ept exceptions, but the way cloak is implemented on intel there is no speed decrease (only for reading, not for executing) so a limited range won't make much sense (and might actually slow it down)

On AMD though it's not possible, and there it will be slow.  My only suggestion on there is to pick a region that's not executed that often.

**Comment by DarkAsm** [comment]
Yes, I just tested it on the intel platform, and the intel platform will not freeze. Thank you for your reply.

**Comment by DarkAsm** [comment]
> On intel it might be possible to implement, as it has sectioned ept exceptions, but the way cloak is implemented on intel there is no speed decrease (only for reading, not for executing) so a limited range won't make much sense (and might actually slow it down)
> 
> On AMD though it's not possible, and there it will be slow. My only suggestion on there is to pick a region that's not executed that often.

Yes, I just tested it on the intel platform, and the intel platform will not freeze. Thank you for your reply.
